test/vapor/config/env_test.exs

```diff
@@ -0,0 +1,31 @@
+defmodule Vapor.Config.EnvTest do
+  use ExUnit.Case, async: false
+
+  alias Vapor.Config.Env
+
+  test "with_prefix/1 loads all env variables with a given prefix" do
+    System.put_env("APP_FOO", "foo")
```

Comment:
So, this is a admittedly a bit of a nit, but I think in these examples, I would prefer to see the key and the values be different from each other. It took be a couple read-throughs to realize that the `"foo"` in `envs["foo"]` is semantically different than the `"foo"` in `== "foor"`.
```suggestion
    System.put_env("APP_FOO", "goo")
```